src/downloader.c

```diff
@@ -1597,11 +1600,8 @@ static void recover_shards(uv_work_t *work)
 
 static void queue_recover_shards(storj_download_state_t *state)
 {
-    if (state->recovering_shards) {
-        return;
-    }
-
-    if (state->pointers_completed) {
+    if (!state->recovering_shards && state->pointers_completed) {
+        state->recovering_shards = true;
```

Comment:
`queue_recover_shards` is in the main thread and nothing else modifies state while running.